Library/Homebrew/utils/analytics.sh

```diff
@@ -1,30 +1,45 @@
 setup-analytics() {
-  # User UUID file. Used for Homebrew user counting. Can be deleted and
-  # recreated with no adverse effect (beyond our user counts being inflated).
-  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  local git_config_file="$HOMEBREW_REPOSITORY/.git/config"
+
+  # Migrate analytics UUID to its new home in Homebrew repo's git config and
+  # remove the legacy UUID file if detected.
+  migrate_legacy_uuid_file () {
```

Comment:
I agree. When I proposed this, I was thinking about making it a top-level function. This has the disadvantage that we have to redefine `git_config_file="$HOMEBREW_REPOSITORY/.git/config"`, but I think that's better than relying on the somewhat obscure access to the local variable in the surrounding function.

Also, for consistency with other shell function names, it would be good to replace underscores with dashes, i.e. make that `migrate-legacy-uuid-file`, and avoid the space between the name and the parentheses.